backend/src/services/episode/episode.service.ts

```diff
@@ -290,9 +302,15 @@ class Episode {
       });
     }
 
+    const favoriteEpisodes = await this.#userFavouriteEpisodeService.getAllByEpisodeId(id);
+    const hasFavoriteEpisodes = Boolean(favoriteEpisodes.length);
+    if(hasFavoriteEpisodes){
+      await this.#userFavouriteEpisodeService.deleteAllByEpisodeId(id);
+    }
```

Comment:
```suggestion
    await this.#userFavouriteEpisodeService.deleteAllByEpisodeId(id);
```
Your realization delete All yEpisodeId allow to use it without any checks, in this case checks will be redundant.